src/overseer/store/jdbc.clj

```diff
@@ -0,0 +1,253 @@
+(ns ^:no-doc overseer.store.jdbc
+  "Implementation of overseer.core/Store for JDBC"
+  (:require [clojure.java.jdbc :as j]
+            [clojure.set :as set]
+            [clojure.string :as string]
+            (clj-time
+              [core :as clj-time]
+              [jdbc]) ; Requiring extends necessary coercion protocols
+            [framed.std.core :as std]
+            [honeysql.core :as sql]
+            (loom graph)
+            (overseer
+              [core :as core]
+              [util :as util])))
+
+(def status-code
+  {:unstarted 0
+   :started 1
+   :finished 2
+   :failed 3
+   :aborted 4})
+
+(def status
+  (set/map-invert status-code))
+
+(defn job->jdbc-map [job]
+  {:pre [job]}
+  (let [{:keys [job/id job/type job/args job/status
+                job/failure job/heartbeat]} job]
+    (-> {:id id
+         :type (name type)
+         :status (get status-code status)}
+        (std/when-assoc :args (some-> args std/to-edn))
+        (std/when-assoc :failure (some-> failure std/to-edn))
+        (std/when-assoc :heartbeat heartbeat))))
+
+(defn jdbc-map->job [jdbc-map]
+  {:pre [jdbc-map]}
+  (-> (std/map-kv #(keyword "job" (name %)) identity jdbc-map)
+      (update :job/status (partial get status))
+      (update :job/type keyword)
+      (util/when-update :job/args std/from-edn)
+      (util/when-update :job/failure std/from-edn)))
+
+(defn query-job
+  "Return a Job map for the given job-id; throws if not found"
+  [db-spec job-id]
+  (->> {:select [:*] :from [:overseer_jobs] :where [:= :id job-id]}
+       sql/format
+       (j/query db-spec)
+       first
+       (std/flip select-keys [:id :type :status :args :failure :heartbeat])
+       jdbc-map->job))
+
+(defn query-lock-version [db-spec job-id]
+  (->> {:select [:lock_version] :from [:overseer_jobs] :where [:= :id job-id]}
+       sql/format
+       (j/query db-spec)
+       first
+       :lock_version))
+
+(defn update-job
+  "Update attributes of a single job using optimistic locking; raises an exception
+  if the job is stale
+
+  where-map - map of {keyword-column-name value} to scope update. id/lock_version
```

Comment:
Heh, that's right. Now that you mention it, the docs are out of date now that it returns instead of throwing! On it!